signer/tests/integration/transaction_signer.rs

```diff
@@ -331,3 +345,232 @@ pub async fn assert_should_be_able_to_handle_sbtc_requests() {
+        signer_private_key: setup.signers.signer.keypair.secret_key().into(),
+        threshold: 2,
+        rng: rand::rngs::StdRng::seed_from_u64(51),
+        dkg_begin_pause: None,
+    };
+
+    // We need to convince the signer event loop that it should accept the
+    // message that we are going to send it.
+    let report = MsgChainTipReport {
+        sender_is_coordinator: true,
+        chain_tip_status: ChainTipStatus::Canonical,
+        chain_tip: BitcoinBlockHash::from([0; 32]),
+    };
+
+    // The message that we will send is for the following sighash. We'll
+    // need to make sure that it is in our database first
+    let sighash_message = [1; 32];
+    let row = BitcoinTxSigHash {
+        txid: BitcoinTxId::from([0; 32]),
+        chain_tip: BitcoinBlockHash::from([0; 32]),
+        prevout_txid: BitcoinTxId::from([0; 32]),
+        prevout_output_index: 0,
+        sighash: SigHash::from(bitcoin::TapSighash::from_byte_array(sighash_message)),
+        prevout_type: model::TxPrevoutType::Deposit,
+        validation_result: signer::bitcoin::validation::InputValidationResult::Ok,
+        is_valid_tx: true,
+        will_sign: true,
+    };
+
+    db.write_bitcoin_txs_sighashes(&[row]).await.unwrap();
+
+    // Now for the nonce request message
+    let mut nonce_request_msg = WstsMessage {
+        txid: bitcoin::Txid::all_zeros(),
+        inner: wsts::net::Message::NonceRequest(NonceRequest {
+            dkg_id: 1,
+            sign_id: 1,
+            sign_iter_id: 1,
+            message: sighash_message.to_vec(),
+            signature_type: wsts::net::SignatureType::Schnorr,
+        }),
+    };
+    let msg_public_key = PublicKey::from_private_key(&PrivateKey::new(&mut rng));
+
+    // Sanity check, the state machines cache should be empty.
+    assert!(tx_signer.wsts_state_machines.is_empty());
+
+    tx_signer
+        .handle_wsts_message(
+            &nonce_request_msg,
+            &report.chain_tip,
+            msg_public_key,
+            &report,
+        )
+        .await
+        .unwrap();
+
+    // We should have a state machine associated with the sighash nonce
+    // request message that we just received.
+    let id1 = StateMachineId::new(sighash_message);
+    assert!(tx_signer.wsts_state_machines.contains(&id1));
+    assert_eq!(tx_signer.wsts_state_machines.len(), 1);
+
+    // Now let's see what happens when we receive a nonce request message
+    // for a sighash that we do not know about. Since the nonce request is
+    // not in the database we should return an error, and the state machine
+    // should not be in the local cache.
+    let random_message: [u8; 32] = Faker.fake_with_rng(&mut rng);
+    match &mut nonce_request_msg.inner {
+        wsts::net::Message::NonceRequest(NonceRequest { message, .. }) => {
+            *message = random_message.to_vec();
+        }
+        _ => panic!("You forgot to update the variant"),
```

Comment:
lolwut?

Reply:
lol I just didn't know what to say here. I was thinking that if you see this panic, then you've updated the message variant and forgot to update this code here to catch the proper variant, or something like that.